YAXLibTests/SerializerOptionsTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
+// Licensed under the MIT license.
+
+using System.Xml.Linq;
+using NUnit.Framework;
+using YAXLib;
+using YAXLib.Options;
+
+namespace YAXLibTests
+{
+    [TestFixture]
+    public class SerializerOptionsTests
+    {
+        private SerializerOptions GetDefaultOptions()
+        {
+            return new();
+        }
+
+        private SerializerOptions GetNonDefaultOptions()
+        {
+            return new() {
+                ExceptionBehavior = YAXExceptionTypes.Ignore,
+                ExceptionHandlingPolicies = YAXExceptionHandlingPolicies.DoNotThrow,
+                AttributeName = new YAXAttributeName
+                    {Dimensions = "test-dims", RealType = "test-realtype"},
+                Namespace = new YAXNameSpace
+                    {Prefix = "testyaxlib", Uri = XNamespace.Get("https://test.sinairv.com/yaxlib/")},
+                MaxRecursion = 123,
+                SerializationOptions = YAXSerializationOptions.SuppressMetadataAttributes
+            };
+        }
+
+        [Test]
+        public void Get_Set_All_Options_Test()
+        {
+            var opt1 = GetDefaultOptions();
+            var opt2 = GetNonDefaultOptions();
+
+            Assert.IsFalse(opt1.ExceptionBehavior == opt2.ExceptionBehavior);
+            Assert.IsFalse(opt1.ExceptionHandlingPolicies == opt2.ExceptionHandlingPolicies);
+            Assert.IsFalse(opt1.AttributeName.Dimensions == opt2.AttributeName.Dimensions);
+            Assert.IsFalse(opt1.Namespace.Prefix == opt2.Namespace.Prefix);
+            Assert.IsFalse(opt1.Namespace.Uri == opt2.Namespace.Uri);
+            Assert.IsFalse(opt1.MaxRecursion == opt2.MaxRecursion);
+            Assert.IsFalse(opt1.SerializationOptions == opt2.SerializationOptions);
+        }
+
+        /// <summary>
+        /// Get YAXSerializer properties which are wrappers to <see cref="SerializerOptions"/>
+        /// </summary>
+        [Test]
+        public void Get_YAXSerializer_WrapperProps_Test()
+        {
+            var opt = GetNonDefaultOptions();
+            var ser = new YAXSerializer(typeof(SerializerOptions), opt);
+
+            Assert.IsTrue(ser.DefaultExceptionType == opt.ExceptionBehavior);
```

Comment:
This isn't really what I meant :)

If the test fails, we will see IsTrue(false) failed , some something like that. Then we can't see on the build server what's wrong and we need to check it locally (same for IsFalse etc)

I would recommend to do:

```c#
Assert.AreEqual (ser.DefaultExceptionType, opt.ExceptionBehavior, "ExceptionBehavior"); 
```

etc. (not 100% about the syntax, not double checked)

If you don't like writing all those description strings, than FluentAssertions helps on that:

```c#
ser.DefaultExceptionType.Should().Be(opt.ExceptionBehavior); // This will give nice  DefaultExceptionType in the message
```

Reply:
Got it, will modify